config/config.go

```diff
@@ -168,7 +167,46 @@ func ReadConfigFile(configFile string, inputPrefix, outputPrefix string) (*Contr
 		return nil, fmt.Errorf("trouble reading config file %q: %w", absConfigFile, err)
 	}
 
-	return ReadConfig(log, yamlFile, inputPrefix, outputPrefix)
+	config, err := ReadConfig(log, yamlFile, inputPrefix, outputPrefix)
+
+	if err != nil {
+		log.Error().Err(err).Msg("could not parse config file")
+		return nil, fmt.Errorf("could not parse config file %q: %w", absConfigFile, err)
+	}
+
+	items, err := ioutil.ReadDir(util.AbsolutePath(inputPrefix, configDir))
+	if err != nil {
+		log.Error().Err(err).Msg("Error reading directory")
+		return nil, fmt.Errorf("could not read config directory %q: %w", configDir, err)
+	}
+	for _, item := range items {
+		fileExtension := filepath.Ext(item.Name())
+		if (fileExtension == ".yaml" || fileExtension == ".yml") && !item.IsDir() {
+
+			currentConfigFile := util.AbsolutePath(inputPrefix, configDir) + "/" + item.Name()
+			log.Debug().Msg("reading config file %q", currentConfigFile)
+			yamlFile, err := ioutil.ReadFile(currentConfigFile)
+			if err != nil {
+				log.Error().Err(err).Msg("could not read config file")
+				return nil, fmt.Errorf("trouble reading config file %q: %w", currentConfigFile, err)
+			}
+
+			currentConfig, err := ReadConfig(log, yamlFile, inputPrefix, outputPrefix)
+			config.VaultConfig.Secrets = append(config.VaultConfig.Secrets, currentConfig.VaultConfig.Secrets...)
```

Comment:
I think the behavior of this is a bit ambiguous. For example: this will add secret/templates/ssh-certs/aws-creds to the internal config but not top level config like the ConfigVersion.

I think this behavior makes sense, however we should explicitly document it (probably in the config dir flag help message)

Reply:
gotcha, I'll add more info in the flag help